src/native/managed/cdacreader/Microsoft.Diagnostics.DataContractReader.Contracts/Contracts/CodeVersions_1.cs

```diff
@@ -223,45 +283,62 @@ private bool IsActiveNativeCodeVersion(NativeCodeVersionHandle nativeCodeVersion
             MethodDescVersioningStateFlags flags = (MethodDescVersioningStateFlags)versioningState.Flags;
             return flags.HasFlag(MethodDescVersioningStateFlags.IsDefaultVersionActiveChildFlag);
         }
-        else if (nativeCodeVersion.CodeVersionNodeAddress != TargetPointer.Null)
+        else
         {
             // NativeCodeVersionNode::IsActiveChildVersion
             Data.NativeCodeVersionNode codeVersion = _target.ProcessedData.GetOrAdd<Data.NativeCodeVersionNode>(nativeCodeVersion.CodeVersionNodeAddress);
             return ((NativeCodeVersionNodeFlags)codeVersion.Flags).HasFlag(NativeCodeVersionNodeFlags.IsActiveChild);
         }
-        else
-        {
-            throw new ArgumentException("Invalid NativeCodeVersionHandle");
-        }
     }
 
-    private NativeCodeVersionHandle FindActiveNativeCodeVersion(ILCodeVersionHandle methodDefActiveVersion, TargetPointer methodDescAddress)
+    private void GetModuleAndMethodDesc(TargetPointer methodDesc, out TargetPointer module, out uint methodDefToken)
+    {
+        IRuntimeTypeSystem rts = _target.Contracts.RuntimeTypeSystem;
+        MethodDescHandle md = rts.GetMethodDescHandle(methodDesc);
+        TargetPointer mtAddr = rts.GetMethodTable(md);
+        TypeHandle typeHandle = rts.GetTypeHandle(mtAddr);
+        module = rts.GetModule(typeHandle);
+        methodDefToken = rts.GetMethodToken(md);
+    }
+
+    private static bool IsExplicit(ILCodeVersionHandle handle)
```

Comment:
I effectively reimplement `IsExplicit` and `IsNode` in the `ReJIT_1.cs` contract and having these helpers on the handles themselves would prevent this. I was hesitant to do so initially because, as you mention, we don't have an explicit notion of public facing vs internal in the handles.

If you are supportive, I'll make that change.

Reply:
These might be a bit weird to add on the contracts themselves. If we add them, then we need to reference the `Target` in the `ILCodeVersionHandle` and we can't version that portion.

Maybe `ILCodeVersionHandle` and `NativeCodeVersionHandle` can hold references to the target but I'm wary because it leaks a lot of state to the contract interface.

After trying this approach, I think it we can move `IsExplicit` to the handle, but should leave `AsNode` as a helper method in the contract implementation.